### PR TITLE
deps/tooling: Bump `envoy.code.check` -> 0.5.5

### PR DESCRIPTION
**Pull request by phlax**
with added `status_upstream` setting

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
cc @alyssawilk this should break when it runs in CI as the existing metadata doesnt have the `status_upstream`

**Comment by phlax** [comment]
i think it failed for the wrong reasons ...

```console
Value has wrong type or shape for Type typing.Dict[str, envoy.code.check.typing.ExtensionMetadataDict]:
```

i think i have made it required in the type check ill update

**Comment by phlax** [comment]
great working!!!

```console
CodeChecker NOTICE [shellcheck] Checks (163) completed successfully
CodeChecker ERROR ERRORS Summary [extensions_metadata]:
--------------------------------------------------------------------------------
You must set (envoy.filters.http.admission_control) `status_upstream` for extensions that are part of `envoy.filters.http.upstream`
You must set (envoy.filters.http.buffer) `status_upstream` for extensions that are part of `envoy.filters.http.upstream`
You must set (envoy.filters.http.upstream_codec) `status_upstream` for extensions that are part of `envoy.filters.http.upstream`
You must set (envoy.filters.http.header_mutation) `status_upstream` for extensions that are part of `envoy.filters.http.upstream`
```

@alyssawilk one of us is going to have to fill in all the blanks

once this is landed i can look at docs etc

**Comment by alyssawilk** [comment]
envoy.filters.http.buffer is super simple and should not be a problem  this can be stable/GA
envoy.filters.http.upstream_codec is the filter which makes upstream proxying work and is stable/GA

envoy.filters.http.header_mutation looks like it tweaks headers but doesn't change the route so should be fine but I'd appreciate @wbpcode chiming in.
envoy.filters.http.admission_control is complicated and I don't know if anyone's using it so I'd make it alpha until someone verifies use in prod.

File: source/extensions/extensions_metadata.yaml [hunk 1]
```diff
@@ -532,6 +535,7 @@ envoy.filters.http.header_mutation:
   - envoy.filters.http.upstream
   security_posture: unknown
   status: alpha
+  status_upstream: alpha
```
Comment (phlax) [inline source/extensions/extensions_metadata.yaml:538]: @wbpcode i think this where you need to chime in - i set to `alpha` for now as that is the general status - but i think we can now declare them separately

@alyssawilk just to confirm that is correct - something can be `alpha` for downstream and `stable` upstream ?

Comment (alyssawilk) [reply source/extensions/extensions_metadata.yaml:538]: oh if it's alpha for downstream lets leave it alpha for upstream too.
It's possible to be stable one way and not the other but in this case I don't think it's warranted.

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**